overlord/snapstate/autorefresh.go

```diff
@@ -172,20 +172,37 @@ func (m *autoRefresh) refreshScheduleWithDefaultsFallback() (ts []*timeutil.Sche
 		return nil, "managed", nil
 	}
 
-	refreshScheduleStr := defaultRefreshSchedule
 	tr := config.NewTransaction(m.state)
-	err = tr.Get("core", "refresh.schedule", &refreshScheduleStr)
+	err = tr.Get("core", "refresh.timer", &scheduleAsStr)
```

Comment:
Sorry for nitpicking over this. I was wondering if it could we written in a way that is more "linear", i.e. where one reading it needs to keep as little state in the head as possible. I came up with: https://paste.ubuntu.com/26409441/ (diff: https://paste.ubuntu.com/26409442/).

[side-note: I also wrote a version that extracts some bits into functions (https://paste.ubuntu.com/26409472/) but that feels wrong because the functions swallow errors and that is not good).

Reply:
Thanks. The code is cleaner. However, it has an unintended consequence of falling back to the legacy setting if the new one is incorrect. I'll tweak your proposal slightly and push an update.